src/xPDO/Om/mysql/xPDOGenerator.php

```diff
@@ -68,21 +68,30 @@ public function getIndex($index) {
      * specified tablePrefix; if tablePrefix is empty, this is ignored.
      * @return boolean True on success, false on failure.
      */
-    public function writeSchema($schemaFile, $package= '', $baseClass= '', $tablePrefix= '', $restrictPrefix= false) {
-        if (empty ($package))
-            $package= $this->manager->xpdo->package;
-        if (empty ($baseClass))
-            $baseClass= 'xPDO\Om\xPDOObject';
-        if (empty ($tablePrefix))
-            $tablePrefix= $this->manager->xpdo->config[xPDO::OPT_TABLE_PREFIX];
+    public function writeSchema(string $schemaFile, string $package = '', string $baseClass = '', string $tablePrefix = '', bool $restrictPrefix = false): bool
+    {
+        if (empty($package)) {
+            $package = $this->manager->xpdo->package;
+        }
+        if (empty($baseClass)) {
+            $baseClass = 'xPDO\Om\xPDOObject';
```

Comment:
Wouldn't it be better to use class name resolution via ::class?

Reply:
This is just for writing the string into the XML. I mean, it could be done that way for the default, I suppose. But I'll consider that in a separate change.